### PR TITLE
#3243 - Reset confirmation when reset password

### PR DESCRIPTION
**Pull request by IrinaZhadzinets**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3243

**Problem:**
* If the user has not yet confirmed the account, but has reset the password, verification is still required, which is inconsistent with behavior in Luma

**In this PR:**
* Changing the password also resets the need to confirmed customer account

### CODE REVIEW

File: src/Model/Resolver/ResetPassword.php [hunk 1]
```diff
@@ -36,12 +38,25 @@ class ResetPassword implements ResolverInterface {
...
     public function __construct(
         Session $customerSession,
-        AccountManagementInterface $accountManagement
+        AccountManagementInterface $accountManagement,
+        ConfirmCustomerByToken $confirmByToken = null
     ) {
         $this->session = $customerSession;
         $this->accountManagement = $accountManagement;
+        $this->confirmByToken = $confirmByToken
+            ?? ObjectManager::getInstance()->get(ConfirmCustomerByToken::class);
```
Comment (zans-laksa) [inline src/Model/Resolver/ResetPassword.php:59]: Can `$this->confirmByToken` ever be `null`?

**Review by zans-laksa — APPROVED** [review]

**Review by zans-laksa — CHANGES_REQUESTED** [review]
.

File: src/Model/Resolver/ResetPassword.php (hunk 1 above)
Comment (IrinaZhadzinets) [reply src/Model/Resolver/ResetPassword.php:59]: I did not get the value as null in this resolver, but it is used this way everywhere in default Magento

Comment (zans-laksa) [reply src/Model/Resolver/ResetPassword.php:70]: The code after the `??` operator will never be called if you run `magento setup:di:compile`.
Please remove it.

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**